crates/rattler_conda_types/src/repo_data/topological_sort.rs

```diff
@@ -158,10 +133,12 @@ fn break_cycles<T: AsRef<PackageRecord>>(
             // prefer arch packages over noarch packages
             let p1_noarch = p1.as_ref().noarch.is_none();
             let p2_noarch = p2.as_ref().noarch.is_none();
+
             if p1_noarch && !p2_noarch {
                 cycle_breaks.insert((pi1.clone(), pi2.clone()));
                 break;
-            } else if !p1_noarch && p2_noarch {
+            } else if !p1_noarch && p2_noarch || i == cycle.len() - 1 {
```

Comment:
Note the added condition here to also return a cycle break if none of the packages is noarch and we're in the last iteration.